control_client/src/main.c

```diff
@@ -96,127 +98,142 @@ int main(int argc, char **argv) {
         }
     }
 
-    int err;
-    err = pad_init(&pad, "127.0.0.1", port);
-    if (err) {
-        fprintf(stderr, "Could not initialize pad server with error: %s\n", strerror(err));
-        exit(EXIT_FAILURE);
-    }
-
-    /* Handle disconnects */
     signal(SIGINT, handle_term);
```

Comment:
This should not be called until after `pad_init`, eve though you did set the socket to `-1`. See my other comment that
the loop should only be for `pad_connect`.